src/observable-map.ts

```diff
@@ -1,7 +1,8 @@
 import { OnHandler, OnChangeHandler, Subscription, ObservableMap, Handlers } from './types';
 
 export const createObservableMap = <T extends { [key: string]: any }>(
-  defaultState?: T
+  defaultState?: T,
+  shouldUpdate: (newV: any, oldValue, prop: keyof T) => boolean = (a,b) => a!==b
```

Comment:
Can it be `prop: keyof T, newValue: any, oldValue: any`? For consistency with set callbacks.